File: Leetcode/1436. Destination City/solution.go
func destCity(paths [][]string) string {
	visited := make(map[string]string, len(paths)*2) // New empty set

	for _, cities := range paths {
		visited[cities[0]] = "s"
		if _, ok := visited[cities[1]]; !ok {
			visited[cities[1]] = ""
		}
	}

	// Get the difference (end - start)
	for city, value := range visited {
		if value == "" {
			return city
		}
	}

	return ""
}

File: Leetcode/1436. Destination City/solution.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        cities = {}
        for path in paths:

            if path[0] not in cities:
                cities[path[0]] = "s"
            else:
                cities[path[0]] = "s"

            if path[1] not in cities:
                cities[path[1]] = None

        final_city = [key for key, value in cities.items() if not value][0]

        return final_city
